### DISCUSSION TITLE
Success with JK_B1A16S Balancer

### DISCUSSION THREAD

**Discussion by delboy711** [syssi/esphome-jk-bms#368] [General]
Another device for the supported list.

Balancer JK_B1A16S
Hardware version V11.5
Software Version  V11.261
![IMG_20230830_151030_500](https://github.com/syssi/esphome-jk-bms/assets/3176010/a759d78b-ba23-4a1f-9936-09ec198a7b6e)

The board is a 24S board depopulated to support just 16 cells.
The CPU is CH32F103 a Chinese clone of an STM32.
The board has a shiny non conformal coating making it impossible to read the labels on the other chips.

Success with BLE protocol version JK02
Everything works.

As for RS485 there is no RS485 interface, but there are three groups of pinouts, so I went hunting to find out what they do.

The group of 4 pins at the rear of the PCB near the CPU were identified as an ST Link programming interface.

At the front edge of the board between the two cell interface sockets are two pins on the left and three pins on the right.

The two pins on the left are connected to GND and PA0 on the CPU (pin 10).  I was hoping this might be a balance enable input. There is a pullup to 3.3V on PA0, but nothing happens when it is shorted to ground.

The group of three pins looked more interesting. The right hand pin appears to be a common, and there are TVS diodes between this pin and the other two.  This implies these pins are used for a remote interface where protection might be required against stray voltages (like an RS485 interface).
![2023-08-27-134813](https://github.com/syssi/esphome-jk-bms/assets/3176010/a328b8f5-cf3f-49a3-bcc2-05aeccf426c2)

I traced these pins to an unpopulated 8 pin IC Pad on the bottom side of the board. They connect to pins 6 and 7 on the pad which happens to the the pins used for the A/B wires of most RS485 chips (like MAX485).
Pins 5 and 8 of the IC pad connect to power, Pins 2 and 3 are commoned together as is the usual connection for the enable pins on an RS485 chip.  That leaves pins 1 and 4 which are the Tx and Rx pins on an RS485 chip.
![2023-08-30-162012](https://github.com/syssi/esphome-jk-bms/assets/3176010/3f3032f5-067a-44a5-a399-ac8876c5a0f8)

Unfortunately there is no connection between pins 1 and 4 and the CPU. They appear to go off somewhere around the other two unpopulated IC pads in the area So it does not seem to be possible to get RS485 working by simply soldering on a MAX485 chip in this location.

I will continue to poke about and try to get RS485 working since I would prefer the reliability of a physical connection over Bluetooth.

**Comment by 23900** [comment]
I consulted the original manufacturer and they said RS485 can only read data and cannot change settings. Settings can only be changed via Bluetooth.
So I'm trying to connect using Bluetooth's UARL.
